lib/Conversion/TritonGPUToLLVM/ReduceOpToLLVM.cpp

```diff
@@ -170,37 +179,47 @@ struct ReduceOpConversion
     ReduceOpHelper helper(op);
     Location loc = op->getLoc();
     unsigned axis = op.getAxis();
+    // Specifies whether the reduce operation returns an index
+    // rather than a value, e.g. argmax, argmin, .. etc
     bool withIndex = triton::ReduceOp::withIndex(op.getRedOp());
 
     auto srcTy = op.getOperand().getType().cast<RankedTensorType>();
     auto srcLayout = srcTy.getEncoding();
     if (!helper.isSupportedLayout()) {
       assert(false && "Unexpected srcLayout in ReduceOpConversion");
     }
+    // The order of the axes for the the threads within the warp
     auto srcOrd = triton::gpu::getOrder(srcLayout);
+    // The elements owned by each thread, e.g. [2, 2] means that each thread
+    // in a warp would own 2x2 elements of the original tensor
     auto sizePerThread = triton::gpu::getSizePerThread(srcLayout);
     auto srcShape = srcTy.getShape();
 
     auto llvmElemTy = getTypeConverter()->convertType(srcTy.getElementType());
     auto llvmIndexTy = getTypeConverter()->getIndexType();
     auto elemPtrTy = LLVM::LLVMPointerType::get(llvmElemTy, 3);
     auto indexPtrTy = LLVM::LLVMPointerType::get(llvmIndexTy, 3);
+    // The shared memory base address
```

Comment:
I don't think this comment is very useful, it just repeats the name of the function below

Reply:
Yea I agree.